pinot-core/src/main/java/org/apache/pinot/core/operator/filter/OrFilterOperator.java

```diff
@@ -60,4 +62,25 @@ public String toExplainString() {
   public List<Operator> getChildOperators() {
     return new ArrayList<>(_filterOperators);
   }
+
+  @Override
+  public boolean canOptimizeCount() {
+    boolean allChildrenProduceBitmaps = true;
+    for (BaseFilterOperator child : _filterOperators) {
+      allChildrenProduceBitmaps &= child.canProduceBitmaps();
+    }
+    return allChildrenProduceBitmaps;
+  }
+
+  @Override
+  public int getNumMatchingDocs() {
```

Comment:
(minor) Add an `assert _filterOperators.size() > 1` for safety and readability

Reply:
Same as and, was handled, no safety problem, indicates a problem elsewhere in the planner.